### ISSUE TITLE
Crash bug in prune 

### ISSUE THREAD

**Issue by mfeif** [restic/restic#2343]
Output of `restic version`
--------------------------
restic 0.9.2 compiled with go1.10.4 on linux/amd64

How did you run restic exactly?
-------------------------------
`restic -vvvvv prune`

(with the following environment variables:)

```
XDG_CACHE_HOME=/mnt/backups/manual/restic_caches
RESTIC_PASSWORD_FILE="/etc/restic/restic-pw.txt"
B2_ACCOUNT_ID="{private stuff}"
B2_ACCOUNT_KEY="{more private stuff}"
B2_CONNECTIONS=10
RESTIC_REPOSITORY="b2:dude-backups-automatic"
RETENTION_DAYS=14
RETENTION_WEEKS=1
RETENTION_MONTHS=12
RETENTION_YEARS=5
BACKUP_PATHS="/mnt/backups/automatic"
BACKUP_EXCLUDES="--exclude-file /mnt/backups/automatic/.backup_exclude"
BACKUP_TAG=systemd.timer
```

What backend/server/service did you use to store the repository?
----------------------------------------------------------------
Backblaze B2

Expected behavior
-----------------
Restic is reporting that my repo has errors or is damaged, and suggests running 'prune':

```
# restic -vvv check
pack 4b0119b9: not referenced in any index
pack c6dd3096: not referenced in any index
pack 6d1feec0: not referenced in any index
pack a8559f97: not referenced in any index
{ many hundreds of similar messages }
pack 4c25ee20: not referenced in any index
pack 9464bd60: not referenced in any index
pack aca11b78: not referenced in any index
62435 additional files were found in the repo, which likely contain duplicate data.
62435 additional files were found in the repo, which likely contain duplicate data.
You can run `restic prune` to correct this.
check snapshots, trees and blobs
error for tree bbf3cce6:
  tree bbf3cce6b4e9f127b04a0c226ce4fd4d2fb8a5962f77f67f06ce827218589115 not found in repository
error for tree 86f4ea98:
  tree 86f4ea9847c8662f7314e93b2ac95cfc18819061d206c330459ef91a51684dda not found in repository
error for tree 9e75689a:
  tree 9e75689a13b7be36a60b2bfc347415d2ddd90fb6b77bc7852ddec37001515d14 not found in repository
error for tree ab8d0a26:
  tree ab8d0a262700d457997b95fb9beaf48e684b9a5ae3df715a0fd4bee5fbb270de not found in repository
error for tree 77b4cad7:
  tree 77b4cad7dfc01f7473d47e62e0150c40f859ae6f60708fcfe17512550640625d not found in repository
error for tree 99874edd:
  tree 99874edd1701a33733234c6ee8766135dba00a4d9a6401c0dee5f0ab0b3798d2 not found in repository
Fatal: repository contains errors
```

Actual behavior
---------------

Restic crashes during a 'prune' operation. It produces lots of errors during 'check':

```
# restic -vvvvv prune
repository 29eb829d opened successfully, password is correct
counting files in repo
building new index for repo
[5:13:31] 100.00%  122024 / 122024 packs
repository contains 122024 packs (831980 blobs) with 587.437 GiB
processed 831980 blobs: 266660 duplicate blobs, 205.001 GiB duplicate
load all snapshots
find data that is still in use for 24 snapshots
tree bbf3cce6b4e9f127b04a0c226ce4fd4d2fb8a5962f77f67f06ce827218589115 not found in repository
github.com/restic/restic/internal/repository.(*Repository).LoadTree
        /build/restic-eUvzEK/restic-0.9.2+ds/obj-x86_64-linux-gnu/src/github.com/restic/restic/internal/repository/repository.go:654
github.com/restic/restic/internal/restic.FindUsedBlobs
        /build/restic-eUvzEK/restic-0.9.2+ds/obj-x86_64-linux-gnu/src/github.com/restic/restic/internal/restic/find.go:11
main.pruneRepository
        /build/restic-eUvzEK/restic-0.9.2+ds/obj-x86_64-linux-gnu/src/github.com/restic/restic/cmd/restic/cmd_prune.go:191
main.runPrune
        /build/restic-eUvzEK/restic-0.9.2+ds/obj-x86_64-linux-gnu/src/github.com/restic/restic/cmd/restic/cmd_prune.go:85
main.glob..func18
        /build/restic-eUvzEK/restic-0.9.2+ds/obj-x86_64-linux-gnu/src/github.com/restic/restic/cmd/restic/cmd_prune.go:25
github.com/spf13/cobra.(*Command).execute
        /build/restic-eUvzEK/restic-0.9.2+ds/obj-x86_64-linux-gnu/src/github.com/spf13/cobra/command.go:762
github.com/spf13/cobra.(*Command).ExecuteC
        /build/restic-eUvzEK/restic-0.9.2+ds/obj-x86_64-linux-gnu/src/github.com/spf13/cobra/command.go:852
github.com/spf13/cobra.(*Command).Execute
        /build/restic-eUvzEK/restic-0.9.2+ds/obj-x86_64-linux-gnu/src/github.com/spf13/cobra/command.go:800
main.main
        /build/restic-eUvzEK/restic-0.9.2+ds/obj-x86_64-linux-gnu/src/github.com/restic/restic/cmd/restic/main.go:86
runtime.main
        /usr/lib/go-1.10/src/runtime/proc.go:198
runtime.goexit
        /usr/lib/go-1.10/src/runtime/asm_amd64.s:2361
```

I've been seeing this trace in my logs from systemd. I manually ran the commands, and during 'check' it said that my repo was damaged. So I manually ran prune, and got the same traceback.

The repo is about 630.8GB according to the control panel at B2. The data it's backing up is only about 266G, so prune has been failing for a while. The data changes extremely rarely, and when it does, it's just kilo- or mega- bytes at a time.

Steps to reproduce the behavior
-------------------------------
I can't deduce what's gone wrong. I've looked into some of the snapshots, doesn't seem like there's much wrong.

Do you have any idea what may have caused this?
-----------------------------------------------
I've seen some other bug discussions about memory use; I have an old box, but I watched as 'prune' ran, and it only ever ate about 4% of RAM.

Do you have an idea how to solve the issue?
-------------------------------------------

Did restic help you or made you happy in any way?
-------------------------------------------------
I'm very happy with restic, and came to it after much research and reading. I'm no pro, but I'm grateful to have such good tools.

**Comment by fd0** [comment]
Hey, thanks for raising this issue. I'm fairly confident that `prune` did not destroy the repo, it is written very conservatively and only ever deletes content at the very end.

What I'm seeing from the log messages:
 * Restic is unable to find data: `tree bbf3cce6b4e9f127b04a0c226ce4fd4d2fb8a5962f77f67f06ce827218589115 not found in repository`
 * There are some files in the repo which are not included in any index: `pack 4b0119b9: not referenced in any index`
 
Unfortunately, restic is unable to resolve this itself in a single run of `prune` (we're working on that one), so the next thing you should do is run `restic rebuild-index`. I suspect the data that could not be found is included in the unreferenced files. Then you can run `check` again to verify that now all data can be found. If necessary, then you can run `prune` to remove data that is not needed any more.

Please let me know how it goes!

**Comment by mfeif** [comment]
Thanks! I tried rebuild-index, and it seemed successful. Then check worked too.

I was "pruning" and the power went out in half my city, so I'll have to pick it up later.

Thanks for the quick response. I'm going to change the title of this so it doesn't attract the wrong kind of searches.

**Comment by mfeif** [comment]
I was able to restart a prune... there are many hundreds of lines like this:
```
unable to refresh lock: Delete: b2_download_file_by_name: 404: bucket dude-backups-automatic does not have file: locks/f0e020ae465a52239b4cd09ccf024d3fe2e1c25601276e6ecd57e190ba344814
```

I've got a ticking clock that looks like this:
```
[39:06:17] 10.80%  10781 / 99835 packs rewritten
```

So it's going to take around 400 hours to prune this backup? Does that seem reasonable? My network is about 10-15Mb upload speed, much faster download (standard US cable setup).

**Comment by fd0** [comment]
@mfeif can you check (via the b2 website) how many files there are in the `locks/` directory?

**Comment by mfeif** [comment]
There are 755 locks.

The repo has grown to 860 GB since pruning, with no new backups. I've set my b2 bucket setting to only keep one copy, so it's restic doing something:

<img width="603" alt="Screen Shot 2019-07-23 at 2 49 46 PM" src="https://user-images.githubusercontent.com/160125/61742718-5cac4880-ad59-11e9-8541-755f04f9ee5b.png">

**Comment by fd0** [comment]
Ok, that is one reason why it will take so long. If you can make sure that there's at most one restic instance running right now, you can remove the files in the `locks/` directory. Restic will check all 750 locks every five minutes, which takes a long time for a high-latency remote like b2. Please report back!

**Comment by mfeif** [comment]
Ok, I've deleted them, I'll keep an eye on it.

Why does prune GROW the size of the repo? It's added 100G since yesterday; for a repo that is in total less than 300G.

**Comment by fd0** [comment]
Prune is the only function which permanently removes data, so we've started with a very conservative implementation that first collects still needed data, saves this data to new files to the repo, and only ever deletes data at the very end of the process, provided all uploads have succeeded. The worst that could happen is restic losing data, we don't want that :)

**Comment by mfeif** [comment]
Ok, after 100+ hours of pruning, a couple hundred gigs of bandwidth, and increasing the size of my repo by a couple hundred gigs, restic crashed again:

First, there were several hundred lines like this:
```
Remove(<lock/f0e020ae46>) returned error, retrying after 3.530064958s: Delete: b2_download_file_by_name: 404: bucket dude-backups-automatic does not
 have file: locks/f0e020ae465a52239b4cd09ccf024d3fe2e1c25601276e6ecd57e190ba344814
Remove(<lock/f0e020ae46>) returned error, retrying after 4.635859887s: Delete: b2_download_file_by_name: 404: bucket dude-backups-automatic does not
 have file: locks/f0e020ae465a52239b4cd09ccf024d3fe2e1c25601276e6ecd57e190ba344814
Remove(<lock/f0e020ae46>) returned error, retrying after 7.375702888s: Delete: b2_download_file_by_name: 404: bucket dude-backups-automatic does not
 have file: locks/f0e020ae465a52239b4cd09ccf024d3fe2e1c25601276e6ecd57e190ba344814
Remove(<lock/f0e020ae46>) returned error, retrying after 12.407565245s: Delete: b2_download_file_by_name: 404: bucket dude-backups-automatic does no
t have file: locks/f0e020ae465a52239b4cd09ccf024d3fe2e1c25601276e6ecd57e190ba344814
Remove(<lock/f0e020ae46>) returned error, retrying after 9.367397555s: Delete: b2_download_file_by_name: 404: bucket dude-backups-automatic does not
 have file: locks/f0e020ae465a52239b4cd09ccf024d3fe2e1c25601276e6ecd57e190ba344814
Remove(<lock/f0e020ae46>) returned error, retrying after 18.872521595s: Delete: b2_download_file_by_name: 404: bucket dude-backups-automatic does no
t have file: locks/f0e020ae465a52239b4cd09ccf024d3fe2e1c25601276e6ecd57e190ba344814
h
```

Then this:
```
hash does not match id: want ae22896e96896bb569229bb6ea7cee18579c22327e46da54ce1d82413a9b472a, got 1c7305b4fb4ed2ebcc3eb48215e4e69c99ca8f08[24/1326]
89c55054d2fafb1
github.com/restic/restic/internal/repository.Repack
        /build/restic-eUvzEK/restic-0.9.2+ds/obj-x86_64-linux-gnu/src/github.com/restic/restic/internal/repository/repack.go:35
main.pruneRepository
        /build/restic-eUvzEK/restic-0.9.2+ds/obj-x86_64-linux-gnu/src/github.com/restic/restic/cmd/restic/cmd_prune.go:278
main.runPrune
        /build/restic-eUvzEK/restic-0.9.2+ds/obj-x86_64-linux-gnu/src/github.com/restic/restic/cmd/restic/cmd_prune.go:85
main.glob..func18
        /build/restic-eUvzEK/restic-0.9.2+ds/obj-x86_64-linux-gnu/src/github.com/restic/restic/cmd/restic/cmd_prune.go:25
github.com/spf13/cobra.(*Command).execute
        /build/restic-eUvzEK/restic-0.9.2+ds/obj-x86_64-linux-gnu/src/github.com/spf13/cobra/command.go:762
github.com/spf13/cobra.(*Command).ExecuteC
        /build/restic-eUvzEK/restic-0.9.2+ds/obj-x86_64-linux-gnu/src/github.com/spf13/cobra/command.go:852
github.com/spf13/cobra.(*Command).Execute
        /build/restic-eUvzEK/restic-0.9.2+ds/obj-x86_64-linux-gnu/src/github.com/spf13/cobra/command.go:800
main.main
        /build/restic-eUvzEK/restic-0.9.2+ds/obj-x86_64-linux-gnu/src/github.com/restic/restic/cmd/restic/main.go:86
runtime.main
        /usr/lib/go-1.10/src/runtime/proc.go:198
runtime.goexit
        /usr/lib/go-1.10/src/runtime/asm_amd64.s:2361
```

Then some more of this:
```
Remove(<lock/f0e020ae46>) returned error, retrying after 334.289944ms: Delete: b2_download_file_by_name: 404: bucket dude-backups-automatic does not
 have file: locks/f0e020ae465a52239b4cd09ccf024d3fe2e1c25601276e6ecd57e190ba344814
Remove(<lock/f0e020ae46>) returned error, retrying after 689.799813ms: Delete: b2_download_file_by_name: 404: bucket dude-backups-automatic does not
 have file: locks/f0e020ae465a52239b4cd09ccf024d3fe2e1c25601276e6ecd57e190ba344814
Remove(<lock/f0e020ae46>) returned error, retrying after 1.195179543s: Delete: b2_download_file_by_name: 404: bucket dude-backups-automatic does not
 have file: locks/f0e020ae465a52239b4cd09ccf024d3fe2e1c25601276e6ecd57e190ba344814
Remove(<lock/f0e020ae46>) returned error, retrying after 2.464650727s: Delete: b2_download_file_by_name: 404: bucket dude-backups-automatic does not
 have file: locks/f0e020ae465a52239b4cd09ccf024d3fe2e1c25601276e6ecd57e190ba344814
Remove(<lock/f0e020ae46>) returned error, retrying after 3.53490661s: Delete: b2_download_file_by_name: 404: bucket dude-backups-automatic does not
have file: locks/f0e020ae465a52239b4cd09ccf024d3fe2e1c25601276e6ecd57e190ba344814
Remove(<lock/f0e020ae46>) returned error, retrying after 3.295126216s: Delete: b2_download_file_by_name: 404: bucket dude-backups-automatic does not
 have file: locks/f0e020ae465a52239b4cd09ccf024d3fe2e1c25601276e6ecd57e190ba344814
Remove(<lock/f0e020ae46>) returned error, retrying after 6.007075009s: Delete: b2_download_file_by_name: 404: bucket dude-backups-automatic does not
 have file: locks/f0e020ae465a52239b4cd09ccf024d3fe2e1c25601276e6ecd57e190ba344814
Remove(<lock/f0e020ae46>) returned error, retrying after 11.41752845s: Delete: b2_download_file_by_name: 404: bucket dude-backups-automatic does not
 have file: locks/f0e020ae465a52239b4cd09ccf024d3fe2e1c25601276e6ecd57e190ba344814
Remove(<lock/f0e020ae46>) returned error, retrying after 15.63838838s: Delete: b2_download_file_by_name: 404: bucket dude-backups-automatic does not
 have file: locks/f0e020ae465a52239b4cd09ccf024d3fe2e1c25601276e6ecd57e190ba344814
Remove(<lock/f0e020ae46>) returned error, retrying after 23.958781716s: Delete: b2_download_file_by_name: 404: bucket dude-backups-automatic does no
t have file: locks/f0e020ae465a52239b4cd09ccf024d3fe2e1c25601276e6ecd57e190ba344814
error in cleanup handler: Delete: b2_download_file_by_name: 404: bucket dude-backups-automatic does not have file: locks/f0e020ae465a52239b4cd09ccf0
24d3fe2e1c25601276e6ecd57e190ba344814
```

And then I'm back at console.

**Comment by mfeif** [comment]
And now I can't do anything with that repo:
```
# restic -vvv stats
Fatal: wrong password or no key found
```

**Comment by fd0** [comment]
Uhm, something very odd is going on here. First, `restic prune` does not touch the files in the `keys/` directory, so this should not happen:
```
Fatal: wrong password or no key found
```

Then, the following errors means that restic got data back it did not expect. This may be an error while downloading the data from B2, or it might hint at a memory issue:
```
hash does not match id: want ae22896e96896bb569229bb6ea7cee18579c22327e46da54ce1d82413a9b472a, got 1c7305b4fb4ed2ebcc3eb48215e4e69c99ca8f08[24/1326]
89c55054d2fafb1
```

Honestly, I'm not sure how to proceed here.

**Comment by mfeif** [comment]
Thank you for trying.

**Comment by didorothy** [comment]
I had a similar error with "hash does not match id..." (see below) and in order to fix the issue I had to find the data file that matched the id mentioned and delete the file. This likely means data loss but it allowed the reindex and prune commands to run properly after that. I have no idea if any other damage was caused. I'm going to run another backup to the repository later and check if everything is then OK.

By the way, I am using a custom built REST backend implementation due to an issue I could not resolve with the one provided by restic so my issue may have been introduced by my implementation. (see https://github.com/didorothy/resticserver if you are interested)

Here is the output of the prune command run with 0.9.5 for Windows 64bit (this was run using the filesystem as a backend and not my custom REST backend):

```
repository 14995d93 opened successfully, password is correct
counting files in repo
building new index for repo
[0:01] 100.00%  30461 / 30461 packs
incomplete pack file (will be removed): 0b384bad4feaa6b39cf3e41ce5ebe990996b6f1a5d8df1f47257a8651302d876
incomplete pack file (will be removed): 1efd46a0119b9fe93cd0dceae1cdd127aef51f2a61bf2232c2e510192f3e1f1b
incomplete pack file (will be removed): 1fb2098b8c417d7c928e13f4512ae9a00537e16e5f715dfbebc5601c3badebba
incomplete pack file (will be removed): 24d488a1376840135b04ea21cbff8fc2b9d792578fc3609e58ca0547f37a873d
incomplete pack file (will be removed): 2bcc2afd45b94245d03cae3dfb9022879fa56e40400aec25a1c3e365ca353b7b
incomplete pack file (will be removed): 3448ffce1a45895cb7fe0453d604e63a88247d6d653cfc230f34796da68256fa
incomplete pack file (will be removed): 41cdb2b2e46a4a6a0eb4b7df6f30f371d4a46341e2be8a100cda08632c99a34b
incomplete pack file (will be removed): 58b01c21b81a45100c392e8a15a262e0fcf93c1b768603cca84cb63c7c11fd77
incomplete pack file (will be removed): 5c7ab191b9456c8a25d7f225daff781cfbd1eb444a728ffb9606fbb580cf00fd
incomplete pack file (will be removed): 5e69443cdb1453c8a19f6db12b6fbff1d2567021096f74df3b16a16c49bfc378
incomplete pack file (will be removed): 75007af414ed1d6f4784a0f8b50d454304413e01f9c230569d27c229271ba18e
incomplete pack file (will be removed): 82357c7ae52b778b36fe81f020ec92932117cec8b6fbfe920669fd30040c3a33
incomplete pack file (will be removed): 887a9c230a675c2483146c8389e995506fbfba5b034fb20f89206ad57190d5e4
incomplete pack file (will be removed): 9b84cd405d501196fd3ea97a340fa1075a444483431600e37ad460ef1d301005
incomplete pack file (will be removed): 9c27e30f953f09ed9b77437502587d44fe3b9de1ae504abbabff6ec88d30faab
incomplete pack file (will be removed): a99c5600fe1903b867478476cc06942a9bd290405cb7ed36d970c3bdf8add497
incomplete pack file (will be removed): ac564f68eeceb865aa6a348421018be60b3a7a64791b704c47b3b4bfb72e5a24
incomplete pack file (will be removed): b52fe5aedfe6ea0cea8a208d940a6f4ee8cd2917ff04ce6d80db09c6aa191802
incomplete pack file (will be removed): d2d00d66d9b255256d2e0e002b929cd2af3b67787e82f101a1b9da486cb24168
incomplete pack file (will be removed): e409efd9d864f088937edd94b4ecf12ebfb8600ea2391119b9cf035b01250fe3
incomplete pack file (will be removed): e551801b2f573a65269aaa8ec46ad2cacba647f02619836d570699a1bf8a8546
repository contains 30440 packs (809635 blobs) with 150.673 GiB
processed 809635 blobs: 244520 duplicate blobs, 10.476 GiB duplicate
load all snapshots
find data that is still in use for 1 snapshots
[0:32] 100.00%  1 / 1 snapshots
found 531039 of 809635 data blobs still in use, removing 278596 blobs
will remove 21 invalid files
will delete 2282 packs and rewrite 3555 packs, this frees 23.679 GiB
hash does not match id: want fe4de0a5df821a00274fa9c415b5a0be0ddb2b7f0ba20ea1bb7b0ba016ecc208, got 7a0aedda7309e8a94a8ecd573eead5ede88090eef3a412cab7cad7e5da2aa2e6
github.com/restic/restic/internal/repository.Repack
        /restic/internal/repository/repack.go:34
main.pruneRepository
        /restic/cmd/restic/cmd_prune.go:278
main.runPrune
        /restic/cmd/restic/cmd_prune.go:85
main.glob..func17
        /restic/cmd/restic/cmd_prune.go:25
github.com/spf13/cobra.(*Command).execute
        /restic/vendor/github.com/spf13/cobra/command.go:762
github.com/spf13/cobra.(*Command).ExecuteC
        /restic/vendor/github.com/spf13/cobra/command.go:852
github.com/spf13/cobra.(*Command).Execute
        /restic/vendor/github.com/spf13/cobra/command.go:800
main.main
        /restic/cmd/restic/main.go:86
runtime.main
        /usr/local/go/src/runtime/proc.go:200
runtime.goexit
        /usr/local/go/src/runtime/asm_amd64.s:1337
```

**Comment by gingerlime** [comment]
I'm seeing similar issues with prune and using backblaze b2, running restic inside docker on a Synology NAS

(sorry for the long snippet)

```
$ docker run --env-file=/volume1/Fantastic/scripts/.restic.env --rm -v /volume1/Fantastic/tmp/.restic-cache/:/root/.cache/ restic/restic -r b2:mochi-backup:restic version
restic 0.9.6 compiled with go1.13.4 on linux/amd64
$ docker run --env-file=/path/to/.restic.env --rm -v /path/to/tmp/.restic-cache/:/root/.cache/ restic/restic -r b2
:backup-bucket:restic prune
counting files in repo
building new index for repo
Load(<data/00a6565f5e>, 591, 5004162) returned error, retrying after 357.131936ms: b2_download_file_by_name: 404: bucket backup-bucket does not have file: restic/data/00/00a6565f5e
e0e4df36b73c1de55741296a679cf03e86f2b5ca3b87f45c8d7166
Load(<data/00a6565f5e>, 591, 5004162) returned error, retrying after 660.492892ms: b2_download_file_by_name: 404: bucket backup-bucket does not have file: restic/data/00/00a6565f5e
e0e4df36b73c1de55741296a679cf03e86f2b5ca3b87f45c8d7166
Load(<data/00a6565f5e>, 591, 5004162) returned error, retrying after 920.315446ms: b2_download_file_by_name: 404: bucket backup-bucket does not have file: restic/data/00/00a6565f5e
e0e4df36b73c1de55741296a679cf03e86f2b5ca3b87f45c8d7166
Load(<data/00a6565f5e>, 591, 5004162) returned error, retrying after 1.635001613s: b2_download_file_by_name: 404: bucket backup-bucket does not have file: restic/data/00/00a6565f5e
e0e4df36b73c1de55741296a679cf03e86f2b5ca3b87f45c8d7166
Load(<data/00a6565f5e>, 591, 5004162) returned error, retrying after 1.982055195s: b2_download_file_by_name: 404: bucket backup-bucket does not have file: restic/data/00/00a6565f5e
e0e4df36b73c1de55741296a679cf03e86f2b5ca3b87f45c8d7166
Load(<data/00a6565f5e>, 591, 5004162) returned error, retrying after 3.011308614s: b2_download_file_by_name: 404: bucket backup-bucket does not have file: restic/data/00/00a6565f5e
e0e4df36b73c1de55741296a679cf03e86f2b5ca3b87f45c8d7166
Load(<data/00a6565f5e>, 591, 5004162) returned error, retrying after 6.715255694s: b2_download_file_by_name: 404: bucket backup-bucket does not have file: restic/data/00/00a6565f5e
e0e4df36b73c1de55741296a679cf03e86f2b5ca3b87f45c8d7166
Load(<data/00a6565f5e>, 591, 5004162) returned error, retrying after 6.138576273s: b2_download_file_by_name: 404: bucket backup-bucket does not have file: restic/data/00/00a6565f5e
e0e4df36b73c1de55741296a679cf03e86f2b5ca3b87f45c8d7166
Load(<data/00a6565f5e>, 591, 5004162) returned error, retrying after 9.010955269s: b2_download_file_by_name: 404: bucket backup-bucket does not have file: restic/data/00/00a6565f5e
e0e4df36b73c1de55741296a679cf03e86f2b5ca3b87f45c8d7166
Load(<data/00a6565f5e>, 591, 5004162) returned error, retrying after 26.189373987s: b2_download_file_by_name: 404: bucket backup-bucket does not have file: restic/data/00/00a6565f5
ee0e4df36b73c1de55741296a679cf03e86f2b5ca3b87f45c8d7166
pack file cannot be listed 00a6565f5ee0e4df36b73c1de55741296a679cf03e86f2b5ca3b87f45c8d7166: b2_download_file_by_name: 404: bucket backup-bucket does not have file: restic/data/00/
00a6565f5ee0e4df36b73c1de55741296a679cf03e86f2b5ca3b87f45c8d7166
Load(<data/0258c46a3d>, 591, 4293740) returned error, retrying after 596.012294ms: b2_download_file_by_name: 404: bucket backup-bucket does not have file: restic/data/02/0258c46a3d
b7e1f1f9af098889d2d05f6e5b8bee6c0ad7edca0755b8a5d37c11
Load(<data/0258c46a3d>, 591, 4293740) returned error, retrying after 601.142011ms: b2_download_file_by_name: 404: bucket backup-bucket does not have file: restic/data/02/0258c46a3d
b7e1f1f9af098889d2d05f6e5b8bee6c0ad7edca0755b8a5d37c11
Load(<data/0258c46a3d>, 591, 4293740) returned error, retrying after 757.424518ms: b2_download_file_by_name: 404: bucket backup-bucket does not have file: restic/data/02/0258c46a3d
b7e1f1f9af098889d2d05f6e5b8bee6c0ad7edca0755b8a5d37c11
Load(<data/0258c46a3d>, 591, 4293740) returned error, retrying after 1.756856005s: b2_download_file_by_name: 404: bucket backup-bucket does not have file: restic/data/02/0258c46a3d
b7e1f1f9af098889d2d05f6e5b8bee6c0ad7edca0755b8a5d37c11
Load(<data/0258c46a3d>, 591, 4293740) returned error, retrying after 2.643018794s: b2_download_file_by_name: 404: bucket backup-bucket does not have file: restic/data/02/0258c46a3d
b7e1f1f9af098889d2d05f6e5b8bee6c0ad7edca0755b8a5d37c11

...

Load(<data/fd92973abf>, 591, 4702814) returned error, retrying after 13.49280179s: b2_download_file_by_name: 404: bucket backup-bucket does not have file: restic/data/fd/fd92973abf9502801c36a3657859eb672a1cbda54bef039d7b42d298b6756279
Load(<data/fd92973abf>, 591, 4702814) returned error, retrying after 13.909613864s: b2_download_file_by_name: 404: bucket backup-bucket does not have file: restic/data/fd/fd92973abf9502801c36a3657859eb672a1cbda54bef039d7b42d298b6756279
pack file cannot be listed fd92973abf9502801c36a3657859eb672a1cbda54bef039d7b42d298b6756279: b2_download_file_by_name: 404: bucket backup-bucket does not have file: restic/data/fd/fd92973abf9502801c36a3657859eb672a1cbda54bef039d7b42d298b6756279
[12:26:05] 92.40%  272300 / 294708 packs

repository contains 272221 packs (1308871 blobs) with 1.277 TiB
processed 1308871 blobs: 101527 duplicate blobs, 79.562 GiB duplicate
load all snapshots
find data that is still in use for 19 snapshots
[2:25] 100.00%  19 / 19 snapshots

found 1055975 of 1308871 data blobs still in use, removing 252896 blobs
will remove 0 invalid files
will delete 30712 packs and rewrite 37822 packs, this frees 252.347 GiB
Load(<data/64df2a5584>, 0, 0) returned error, retrying after 689.425709ms: b2_download_file_by_name: 404: bucket backup-bucket does not have file: restic/data/64/64df2a55844d00de76
6dc8ad98a6cb2fec4958b912e22335c37c592c53337520
Load(<data/64df2a5584>, 0, 0) returned error, retrying after 503.347403ms: b2_download_file_by_name: 404: bucket backup-bucket does not have file: restic/data/64/64df2a55844d00de76
6dc8ad98a6cb2fec4958b912e22335c37c592c53337520
Load(<data/64df2a5584>, 0, 0) returned error, retrying after 1.085691857s: b2_download_file_by_name: 404: bucket backup-bucket does not have file: restic/data/64/64df2a55844d00de76
6dc8ad98a6cb2fec4958b912e22335c37c592c53337520
Load(<data/64df2a5584>, 0, 0) returned error, retrying after 1.566190034s: b2_download_file_by_name: 404: bucket backup-bucket does not have file: restic/data/64/64df2a55844d00de76
6dc8ad98a6cb2fec4958b912e22335c37c592c53337520
Load(<data/64df2a5584>, 0, 0) returned error, retrying after 2.772795009s: b2_download_file_by_name: 404: bucket backup-bucket does not have file: restic/data/64/64df2a55844d00de76
6dc8ad98a6cb2fec4958b912e22335c37c592c53337520
Load(<data/64df2a5584>, 0, 0) returned error, retrying after 2.238372523s: b2_download_file_by_name: 404: bucket backup-bucket does not have file: restic/data/64/64df2a55844d00de76
6dc8ad98a6cb2fec4958b912e22335c37c592c53337520
Load(<data/64df2a5584>, 0, 0) returned error, retrying after 3.927939936s: b2_download_file_by_name: 404: bucket backup-bucket does not have file: restic/data/64/64df2a55844d00de76
6dc8ad98a6cb2fec4958b912e22335c37c592c53337520
Load(<data/64df2a5584>, 0, 0) returned error, retrying after 7.664196499s: b2_download_file_by_name: 404: bucket backup-bucket does not have file: restic/data/64/64df2a55844d00de76
6dc8ad98a6cb2fec4958b912e22335c37c592c53337520
Load(<data/64df2a5584>, 0, 0) returned error, retrying after 7.190738495s: b2_download_file_by_name: 404: bucket backup-bucket does not have file: restic/data/64/64df2a55844d00de76
6dc8ad98a6cb2fec4958b912e22335c37c592c53337520
Load(<data/64df2a5584>, 0, 0) returned error, retrying after 28.315969891s: b2_download_file_by_name: 404: bucket backup-bucket does not have file: restic/data/64/64df2a55844d00de7
66dc8ad98a6cb2fec4958b912e22335c37c592c53337520
b2_download_file_by_name: 404: bucket backup-bucket does not have file: restic/data/64/64df2a55844d00de766dc8ad98a6cb2fec4958b912e22335c37c592c53337520
Load
github.com/restic/restic/internal/repository.DownloadAndHash
        /restic/internal/repository/repository.go:786
github.com/restic/restic/internal/repository.Repack
        /restic/internal/repository/repack.go:26
main.pruneRepository
        /restic/cmd/restic/cmd_prune.go:278
main.runPrune
        /restic/cmd/restic/cmd_prune.go:85
main.glob..func18
        /restic/cmd/restic/cmd_prune.go:25
github.com/spf13/cobra.(*Command).execute
        /restic/vendor/github.com/spf13/cobra/command.go:762
github.com/spf13/cobra.(*Command).ExecuteC
        /restic/vendor/github.com/spf13/cobra/command.go:852
github.com/spf13/cobra.(*Command).Execute
        /restic/vendor/github.com/spf13/cobra/command.go:800
main.main
        /restic/cmd/restic/main.go:86
runtime.main
        /usr/local/go/src/runtime/proc.go:203
runtime.goexit
        /usr/local/go/src/runtime/asm_amd64.s:1357
Repack
github.com/restic/restic/internal/repository.Repack
        /restic/internal/repository/repack.go:28
main.pruneRepository
        /restic/cmd/restic/cmd_prune.go:278
main.runPrune
        /restic/cmd/restic/cmd_prune.go:85
main.glob..func18
        /restic/cmd/restic/cmd_prune.go:25
github.com/spf13/cobra.(*Command).execute
        /restic/vendor/github.com/spf13/cobra/command.go:762
github.com/spf13/cobra.(*Command).ExecuteC
        /restic/vendor/github.com/spf13/cobra/command.go:852
github.com/spf13/cobra.(*Command).Execute
        /restic/vendor/github.com/spf13/cobra/command.go:800
main.main
        /restic/cmd/restic/main.go:86
runtime.main
        /usr/local/go/src/runtime/proc.go:203
runtime.goexit
        /usr/local/go/src/runtime/asm_amd64.s:1357
```

besides that, it appears that even when operations terminate successfully, the repository gets locked. I need to very frequently (always?) unlock before/after performing operations ... My connection is pretty solid, so I don't think it's a networking issue.

Anything worth testing/checking?

**Comment by MichaelEischer** [comment]
@mfeif Do you still have the prune issue, if yes please reopen this issue. I'd expect you've deleted the damaged repository by now? We have a new prune implementation on the main development branch which could help with your prune issues.

@didorothy Such damaged packs often indicate hardware problems. If you still have this issue please open a new issue so we can investigate it further.

@gingerlime Do you still have this prune issue? This looks a lot like B2 was temporarily unable to find some files in the repository O.o (or you somehow had two prune runs in parallel???). If you still have this issue please open a new issue so we can investigate it further.

**Comment by gingerlime** [comment]
@MichaelEischer I’m using the master version for a while and it’s fantastic. I also switched to using Backblaze S3 “proxy” although it’s hard to say this had an effect. I’m looking forward to a new official version on docker! Thanks so much for all the amazing work that goes into restic
